webapp/packages/plugin-connection-search/src/Search/ConnectionSearchService.ts

```diff
@@ -156,33 +151,28 @@ export class ConnectionSearchService {
       return;
     }
 
-    if (!this.formState) {
-      this.formState = new ConnectionFormState(
-        this.projectsService,
-        this.projectInfoResource,
-        this.connectionFormService,
-        this.connectionInfoResource,
-        this.connectionInfoOriginResource,
-      );
+    this.formState?.dispose();
+    this.formState?.disposeTask.removeHandler(this.goBack.bind(this));
+    this.formState = new ConnectionFormState(this.serviceProvider, this.connectionFormService, {
+      submitType: null,
+      projectId: projects[0]!.id,
+      availableDrivers: database.possibleDrivers,
+      type: 'public',
+      requiredNetworkHandlersIds: [],
+    });
 
-      this.formState.closeTask.addHandler(this.goBack.bind(this));
-    }
+    runInAction(() => {
+      this.optionsPart!.state.host = database.host;
+      this.optionsPart!.state.port = String(database.port);
+      this.optionsPart!.state.driverId = database.defaultDriver;
+    });
 
-    this.formState
-      .setOptions('create', 'public')
-      .setConfig(projects[0]!.id, {
-        ...this.connectionInfoResource.getEmptyConfig(),
-        driverId: database.defaultDriver,
-        host: database.host,
-        port: `${database.port}`,
-      })
-      .setAvailableDrivers(database.possibleDrivers);
-
-    this.formState.load();
+    this.formState.disposeTask.addHandler(this.goBack.bind(this));
   }
 
   private clearFormState() {
     this.formState?.dispose();
+    this.formState?.disposeTask.removeHandler(this.goBack.bind(this));
```

Comment:
This will not work as expected, even more as we plan to remove the form state object at all; we probably don't need to remove handlers (GC should handle it)